ocaml/libs/pciutil/pciutil.ml

```diff
@@ -62,12 +62,13 @@ let parse vendor device =
       (fun path -> try Unix.access path perms ; true with _ -> false)
       l
   in
-  try
-    (* is that the correct path ? *)
-    let l =
-      access_list
-        ["/usr/share/hwdata/pci.ids"; "/usr/share/misc/pci.ids"]
-        [Unix.R_OK]
-    in
-    parse_from (List.hd l) vendor device
-  with _ -> (unknown_vendor vendor, unknown_device device)
+  access_list
+    ["/usr/share/hwdata/pci.ids"; "/usr/share/misc/pci.ids"]
+    [Unix.R_OK]
+  |> Fun.flip List.nth_opt 0
```

Comment:
I see `List.nth_opt ... 0` being used repeatedly. You could define:
```
let hd xs = List.nth_opt xs 0
let hd = function x::_ -> Some x | _ -> None`
```

or similar.